plugins/woocommerce/src/Internal/Admin/ProductReviews/Reviews.php

```diff
@@ -0,0 +1,601 @@
+		add_action( 'wp_ajax_edit-comment', [ $this, 'handle_edit_review' ], -1 );
+		add_action( 'wp_ajax_replyto-comment', [ $this, 'handle_reply_to_review' ], -1 );
+
+		add_filter( 'parent_file', [ $this, 'edit_review_parent_file' ] );
+
+		add_filter( 'gettext', [ $this, 'edit_comments_screen_text' ], 10, 2 );
+
+		add_action( 'admin_notices', [ $this, 'display_notices' ] );
+	}
+
+	/**
+	 * Gets the required capability to access the reviews page and manage product reviews.
+	 *
+	 * @param string $context The context for which the capability is needed (e.g. `view` or `moderate`).
+	 * @return string
+	 */
+	public static function get_capability( string $context = 'view' ) : string {
+
+		/**
+		 * Filters whether the current user can manage product reviews.
+		 *
+		 * @since 6.6.0
+		 *
+		 * @param string $capability The capability (defaults to `moderate_comments`).
+		 * @param string $context    The context for which the capability is needed.
+		 */
+		return (string) apply_filters( 'woocommerce_product_reviews_page_capability', 'moderate_comments', $context );
+	}
+
+	/**
+	 * Registers the Product Reviews submenu page.
+	 *
+	 * @return void
+	 */
+	public function add_reviews_page() : void {
+
+		$this->reviews_page_hook = add_submenu_page(
+			'edit.php?post_type=product',
+			__( 'Reviews', 'woocommerce' ),
+			__( 'Reviews', 'woocommerce' ) . $this->get_pending_count_bubble(),
+			static::get_capability(),
+			static::MENU_SLUG,
+			[ $this, 'render_reviews_list_table' ]
+		);
+
+		add_action( "load-{$this->reviews_page_hook}", [ $this, 'load_reviews_screen' ] );
+	}
+
+	/**
+	 * Retrieves the URL to the product reviews page.
+	 *
+	 * @return string
+	 */
+	public static function get_reviews_page_url() : string {
+		return add_query_arg(
+			[
+				'post_type' => 'product',
+				'page'      => static::MENU_SLUG,
+			],
+			admin_url( 'edit.php' )
+		);
+	}
+
+	/**
+	 * Determines whether the current page is the reviews page.
+	 *
+	 * @global WP_Screen $current_screen
+	 *
+	 * @return bool
+	 */
+	public function is_reviews_page() : bool {
+		global $current_screen;
+
+		return isset( $current_screen->base ) && 'product_page_' . static::MENU_SLUG === $current_screen->base;
+	}
+
+	/**
+	 * Loads the JavaScript required for inline replies and quick edit.
+	 *
+	 * @return void
+	 */
+	public function load_javascript() : void {
+		if ( $this->is_reviews_page() ) {
+			wp_enqueue_script( 'admin-comments' );
+			enqueue_comment_hotkeys_js();
+		}
+	}
+
+	/**
+	 * Determines if the object is a review or a reply to a review.
+	 *
+	 * @param WP_Comment|array|null $object Object to check.
+	 * @return bool
+	 */
+	protected function is_review_or_reply( $object ) : bool {
+		if ( ! $object instanceof WP_Comment ) {
+			return false;
+		}
+
+		return 'review' === $object->comment_type || 'product' === get_post_type( $object->comment_post_ID );
```

Comment:
What are your thoughts on explicitly checking against both valid comment types—so something like:

```php
return ( 'review' === $object->comment_type || 'comment' === $object->comment_type )
	&& 'product' === get_post_type( $object->comment_post_ID );
```

(Just thinking about cases where extensions/3PDs add further comment types to products that are neither reviews or replies, but cover some other purpose.)

Reply:
Maybe our rationale was looking at it from the way around and assumed that if a third party plugin adds special comment types to products then they would be listed here as well. Either way the return value here could be wrapped to a filter as well to help third parties integrate, what do you think?

Reply:
> Maybe our rationale was looking at it from the way around and assumed that if a third party plugin adds special comment types to products then they would be listed here as well.

If that is the case, couldn't we drop the first part of the check (that is, if the intention is to display *all* comments associated with products, I don't think we need to explicitly check if the comment type is a review)?

> Either way the return value here could be wrapped to a filter as well to help third parties integrate, what do you think?

Yes, we could, though unless there is strong disagreement I'd suggest we don't do that right away (generally, we don't want to expand the public API surface—including via hooks—beyond what is needed, and especially so from within internally namespaced code).

Reply:
> If that is the case, couldn't we drop the first part of the check (that is, if the intention is to display all comments associated with products, I don't think we need to explicitly check if the comment type is a review)?

I'll discuss this with my team and get back to you.

> > Either way the return value here could be wrapped to a filter as well to help third parties integrate, what do you think?
>
> Yes, we could, though unless there is strong disagreement I'd suggest we don't do that right away (generally, we don't want to expand the public API surface—including via hooks—beyond what is needed, and especially so from within internally namespaced code).

See https://github.com/woocommerce/woocommerce/pull/32763/commits/a8d10c488ea159e72733629e9f3d468dd7f69d85 - we may need this in Product Reviews Pro